src/Error/Error.php

```diff
@@ -171,7 +171,7 @@ public static function createLocatedError($error, $nodes = null, $path = null)
             $message = (string) $error;
         }
 
-        return new static(
+        return new self(
```

Comment:
Right. I must say i don't really like the phpstan rule behind this for library code. Locking down classes in very rigid ways hurts reusability and flexibility.

Reply:
@vladar i think the route we choose to go on these issues is an important design decision. There are basically two paths we can go down.

We can lock everything down to the max, making all classes final and methods preferrably private. This does prevent misuse and keeps the API surface small, but hurts flexibility and extensibility. I personally think this approach makes more sense in large project codebases where you want to keep core components clean and limit technical debt.

The other way would be to keep classes open, allowing extending them and overwriting or calling protected methods. The BC guarantees are only upheld for the parts of the library marked with `@api`. While this provides some footguns and may cause pain for developers who choose to mess with the insides of this library, i think that this is not really a drawback from a library perspective. It gives more power and responsibility to users of this library.

Reply:
I agree with you here. I think it is quite an impractical rule. `@api` convention should be enough to communicate which APIs we consider `stable`. Then the risk of overriding something is on developers messing with internals.

I know this feeling when library authors didn't consider a specific use-case and locked-down everything so that you couldn't even "fix" it on the user-land without forking or some crazy reflection hacks.

Reply:
I reverted the breaking changes here for now, we can continue this discussion in https://github.com/webonyx/graphql-php/issues/593